AsyncDisplayKit/UIResponder+AsyncDisplayKit.m

```diff
@@ -0,0 +1,31 @@
+//
+//  UIResponder+AsyncDisplayKit.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 2/13/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import "UIResponder+AsyncDisplayKit.h"
+
+#import <AsyncDisplayKit/ASAssert.h>
+#import <AsyncDisplayKit/ASBaseDefines.h>
+#import <AsyncDisplayKit/ASResponderChainEnumerator.h>
+
+@implementation UIResponder (AsyncDisplayKit)
+
+- (__kindof UIViewController *)asdk_associatedViewController
+{
+  ASDisplayNodeAssertMainThread();
+  
+  for (UIResponder *responder in [self asdk_responderChainEnumerator]) {
+    UIViewController *vc = ASDynamicCast(responder, UIViewController);
```

Comment:
Is this documented behavior? A responder will have it's viewController in the responder chain prior to any other possible view controllers at all times? If not, maybe a comment indicating it's been tested to be true in practice? If so a link would be awesome :)